lndr-backend/src/Lndr/Server.hs

```diff
@@ -239,8 +241,23 @@ verifyIndividualRecord creditHash = do
     let recordNotFound = throwError $
             err404 { errBody = "Credit hash does not refer to pending bilateral settlement record" }
     bilateralCreditRecord <- maybe recordNotFound pure recordM
-    verifySettlementPayment bilateralCreditRecord
+    let txHashNotFound = throwError $
+            err404 { errBody = "Bilateral settlement record does not have a transaction hash" }
+    transHash <- maybe txHashNotFound pure $ txHash bilateralCreditRecord
+    multiSettlementRecords <- liftIO $ withResource pool $ Db.lookupCreditsByTxHash transHash
+    let settlementAmount = foldr getSettlementAmount 0 multiSettlementRecords
+    liftIO $ pushLogStrLn loggerSet . toLogStr $ ("SETTLEMENT AMOUNT TO VERIFY " :: String) ++ (show settlementAmount)
+    verifySettlementPayment bilateralCreditRecord settlementAmount
     liftIO $ withResource pool $ Db.verifyCreditByHash creditHash
     web3Result <- finalizeTransaction configTVar bilateralCreditRecord
                         `catchError` (pure . T.pack . show)
     liftIO $ pushLogStrLn loggerSet . toLogStr . ("WEB3: " ++) . show $ web3Result
+
+
+getSettlementAmount :: BilateralCreditRecord -> Integer -> Integer
```

Comment:
Don't make this a top-level function if you only intend to use it in a fold. That will make the whole file difficult to read.

Instead, do something like: `foldr (\cr sum -> fromMaybe 0 $ settlementAmount cr)`